### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Elerqsousy**
In this milestone:
- Created class classroom
- Added associations between classes book and student
- Created classes Book and Rental
- Added associations between Book, Rental and Person classes

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  attr_accessor :date, :person, :book
+
+  def initialize(book, person, date)
+    @date = date
+    @person = person
+    @book = book
+    book.rentals.push(self)
+    person.rentals.push(self)
+  end
+end
```
Comment (rustam-tolipov) [inline book.rb]: - [x] It appears that the Render class is a separate entity from the Book class and should therefore be placed in a different file  📁. To ensure this, please create a new file 📂✏️ for the Render class and place it there. ✏️🖊️

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 require './nameable'
-
+require './book'
 # Class Person
 class Person < Nameable
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals, :parent_permession
```
Comment (rustam-tolipov) [inline person.rb]: - [x] At that point, I do not think you are going to need `:parent_permession`. Kindly remove it if it's not necessary. 🚮🗑️

File: person.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permession: true)
```
Comment (rustam-tolipov) [inline person.rb]: - [x] Are you sure it's `parent_permession` or is it `parent_permission`? 😊 Kindly fix the typo 📝

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(book, person, date)
```
Comment (rustam-tolipov) [inline book.rb]: - [x] You only need to add your person, and date. And rest of the features need to be handled in the Rental class. Kindly remove `book` from add_rental params.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Elerqsousy,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ha-manel — APPROVED** [review]
# STATUS: APPROVED ✅

## Hi @Elerqsousy👋

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉**



## Highlights:
* Well-written PR description ✔️ 
* All tests passed ✔️ 

## Optional suggestions:
- [x] Kindly add the MIT file to your project because the link in your README file is broken 😓 


------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**